boot/unlocked_state.go

```diff
@@ -0,0 +1,95 @@
+package boot
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+// states for partition state
+const (
+	// states for LocateState
+	PartitionFound      = "found"
+	PartitionNotFound   = "not-found"
+	PartitionErrFinding = "error-finding"
+	// states for MountState
+	PartitionMounted          = "mounted"
+	PartitionErrMounting      = "error-mounting"
+	PartitionAbsentOptional   = "absent-but-optional"
+	PartitionMountedUntrusted = "mounted-untrusted"
+	// states for UnlockState
+	PartitionUnlocked     = "unlocked"
+	PartitionErrUnlocking = "error-unlocking"
+	// keys used to unlock for UnlockKey
+	KeyRun      = "run"
+	KeyFallback = "fallback"
+	KeyRecovery = "recovery"
+)
+
+// partitionState is the state of a partition after recover mode has completed
+// for degraded mode.
+type PartitionState struct {
+	// MountState is whether the partition was mounted successfully or not.
+	MountState string `json:"mount-state,omitempty"`
+	// MountLocation is where the partition was mounted.
+	MountLocation string `json:"mount-location,omitempty"`
+	// Device is what device the partition corresponds to. It can be the
+	// physical block device if the partition is unencrypted or if it was not
+	// successfully unlocked, or it can be a decrypted mapper device if the
+	// partition was encrypted and successfully decrypted, or it can be the
+	// empty string (or missing) if the partition was not found at all.
+	Device string `json:"device,omitempty"`
+	// FindState indicates whether the partition was found on the disk or not.
+	FindState string `json:"find-state,omitempty"`
+	// UnlockState was whether the partition was unlocked successfully or not.
+	UnlockState string `json:"unlock-state,omitempty"`
+	// UnlockKey was what key the partition was unlocked with, either "run",
+	// "fallback" or "recovery".
+	UnlockKey string `json:"unlock-key,omitempty"`
+}
+
+type UnlockState struct {
```

Comment:
this needs a doc comment

Reply:
also this maybe needs to say Disk in its name? as we have already an Unlocker in boot that has nothing to do with this

Reply:
I agree with the name change, also `UnlockState.PartitionState.UnlockState` is a little confusing.